src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -506,90 +506,48 @@ public void Dispose()
         /// Returns the correct colorspace based on the image component count and the jpeg frame component id's.
         /// </summary>
         /// <param name="componentCount">The number of components.</param>
+        /// <param name="adobeMarker">Parsed adobe APP14 marker.</param>
         /// <returns>The <see cref="JpegColorSpace"/></returns>
-        private JpegColorSpace DeduceJpegColorSpace(byte componentCount)
+        internal static JpegColorSpace DeduceJpegColorSpace(byte componentCount, ref AdobeMarker adobeMarker)
         {
-            if (componentCount == 1)
```

Comment:
I have found a jpeg image with an app14 marker which is gray, should that not be possible according to the spec?
![gray-sample](https://user-images.githubusercontent.com/38701097/179360954-1a439874-c60a-4058-9c11-56693e5f29b7.jpg)

Reply:
Adobe spec doesn't say anything about grayscale images but I think we can include it as decoding it doesn't violate the spec :)